### ISSUE TITLE
GPhotoFrame stops changing pictures

### ISSUE THREAD

**Issue by GoogleCodeExporter** [yendo/gphotoframe#94]
```
What steps will reproduce the problem?
1. Nothing, just run the program.
2. After few pictures gphotoframe stops on some picture.
3. Only 'Next photo' from pop-up menu change picture.

What is the expected output? What do you see instead?
changing photos like is set in preferences.

What version of the product are you using? On what operating system?
 1.3. Ubuntu Linux 11.04

Please provide any additional information below.

```

Original issue reported on code.google.com by `qki...@gmail.com` on 14 Jun 2011 at 6:55

**Comment by GoogleCodeExporter** [comment]
```
Could you install the attached gphotoframe debug version (1.4-b2.1) and run 
this in command line?
This version shows debug messeages.

What type of photo source do you set (flickr, folder...) in photo source tab?
Could you try your photo source (or other one) one by one?

```

Original comment by `yendo0206` on 15 Jun 2011 at 5:34

Attachments:
- [gphotoframe_1.4~b2.1-0.1_all.deb](https://storage.googleapis.com/google-code-attachments/gphotoframe/issue-94/comment-1/gphotoframe_1.4~b2.1-0.1_all.deb)

**Comment by GoogleCodeExporter** [comment]
```
OK, I'll install debug version.

GPhotoFrame stops changing photo on any kind of source. I have turned on 
folder, flickr and facebook source of pictures.
```

Original comment by `qki...@gmail.com` on 15 Jun 2011 at 4:49

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
> File "/usr/lib/python2.7/dist-packages/gphotoframe/utils/sqldb.py", line 26, 
in fetchone
>     data = self.db.execute(sql).fetchone()[0]
> sqlite3.OperationalError: near "5": syntax error

sqlite error?

Could you install the attached gphotoframe debug version (1.4-b2.2) and run 
this in command line?
This version shows sql debug messeages.
```

Original comment by `yendo0206` on 19 Jun 2011 at 10:35

Attachments:
- [gphotoframe_1.4~b2.2-0.1_all.deb](https://storage.googleapis.com/google-code-attachments/gphotoframe/issue-94/comment-5/gphotoframe_1.4~b2.2-0.1_all.deb)

**Comment by GoogleCodeExporter** [comment]
```
Debug messages from command line:

<gphotoframe.plugins.folder.DirPhotoList object at 0xacc5e6c>
file:///home/zdjęcia/2006/07/bloom/koncert/IMG_1328.JPG
fetchone: SELECT count(*) FROM ban WHERE 
url='file:///home/zdjęcia/2006/07/bloom/koncert/IMG_1328.JPG';
SELECT id, url FROM photoframe ORDER BY id DESC LIMIT 1;
INSERT INTO photoframe VALUES (2823, 
'file:///home/zdjęcia/2006/07/bloom/koncert/IMG_1328.JPG', '', 'IMG_1328.JPG', 
'', 1153245954.0, 'Folder', '');
INSERT INTO photoframe VALUES (2823, 
'file:///home/zdjęcia/2006/07/bloom/koncert/IMG_1328.JPG', '', 'IMG_1328.JPG', 
'', 1153245954.0, 'Folder', '');
fetchone: SELECT count(*) FROM photoframe
DELETE FROM photoframe WHERE id < (select id FROM photoframe ORDER BY id DESC 
LIMIT 1) - 100
DELETE FROM photoframe WHERE id < (select id FROM photoframe ORDER BY id DESC 
LIMIT 1) - 100
4102 10 True
<gphotoframe.plugins.folder.DirPhotoList object at 0xacc5e6c>
file:///home/zdjęcia/2006/07/Barry's Open Air Festival 2/P1080113.JPG
fetchone: SELECT count(*) FROM ban WHERE 
url='file:///home/zdjęcia/2006/07/Barry's Open Air Festival 2/P1080113.JPG';
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/gphotoframe/liststore.py", line 79, in _start_timer
    updated = self._change_photo()
  File "/usr/lib/python2.7/dist-packages/gphotoframe/liststore.py", line 105, in _change_photo
    photo_list.get_photo(self._show_photo_cb)
  File "/usr/lib/python2.7/dist-packages/gphotoframe/plugins/folder.py", line 47, in get_photo
    cb(None, self.photo)
  File "/usr/lib/python2.7/dist-packages/gphotoframe/liststore.py", line 116, in _show_photo_cb
    if self.ban_db.check_banned_for(photo.get('url')):
  File "/usr/lib/python2.7/dist-packages/gphotoframe/liststore.py", line 228, in check_banned_for
    return self.con.fetchone(sql)
  File "/usr/lib/python2.7/dist-packages/gphotoframe/utils/sqldb.py", line 29, in fetchone
    data = self.db.execute(sql).fetchone()[0]
sqlite3.OperationalError: near "s": syntax error


```

Original comment by `qki...@gmail.com` on 23 Jun 2011 at 8:25

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your reporting.
I fixed this bug in revision 902f43226ecc.
```

Original comment by `yendo0206` on 23 Jun 2011 at 12:24
- Changed state: **Fixed**
